File: Lab4/app/src/main/java/com/example/lab4/MainActivity.java
package com.example.lab4;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        Button foodOrderButton = findViewById(R.id.order_food_button);
        foodOrderButton.setOnClickListener(v -> {
            // Start the FoodOrderActivity when the button is clicked
            startActivity(new Intent(MainActivity.this, FoodOrderActivity.class));
        });

        String foodText = getIntent().getStringExtra("food_text");
        TextView foodTextView = findViewById(R.id.food_text);
        if (foodText != null) {
            foodTextView.setText(foodText);
            foodTextView.setVisibility(View.VISIBLE);
        } else {
            foodTextView.setVisibility(View.GONE);
        }
    }
}

File: Lab8/app/src/main/java/com/example/lab8/MainActivity.kt
package com.example.lab8

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        createNotificationChannel()

        val notifyButton = findViewById<Button>(R.id.notify_button)
        notifyButton.setOnClickListener {
            Toast.makeText(this, "Notification Button Clicked", Toast.LENGTH_SHORT).show()
            showNotification()
        }

    }

    private fun createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            if (checkSelfPermission(android.Manifest.permission.POST_NOTIFICATIONS)
                != PackageManager.PERMISSION_GRANTED
            ) {
                requestPermissions(arrayOf(android.Manifest.permission.POST_NOTIFICATIONS), 1001)
            }
            val channel = NotificationChannel(
                "default_channel",
                "Default Channel",
                NotificationManager.IMPORTANCE_DEFAULT
            )
            val notificationManager =
                getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            notificationManager.createNotificationChannel(channel)
        } else {
            Toast.makeText(
                this,
                "Notifications are not supported on this version",
                Toast.LENGTH_SHORT
            ).show()
        }
    }

    private fun showNotification() {
        val notification = Notification.Builder(this, "default_channel")
            .setContentTitle("Notification Title")
            .setContentText("This is the content of the notification.")
            .setSmallIcon(R.drawable.ic_notification_bell)


        val notificationManager =
            getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        notificationManager.notify(getNotificationId(), notification.build())

    }


    private fun getNotificationId(): Int {
        return System.currentTimeMillis().toInt() % 10000
    }
}